storage/range.go

```diff
@@ -420,11 +422,17 @@ func (r *Range) Desc() *proto.RangeDescriptor {
 	return (*proto.RangeDescriptor)(atomic.LoadPointer(&r.desc))
 }
 
-// SetDesc atomically sets the range's descriptor. This method should
-// be called in the context of having metaLock held, as is the case
-// for merging, splitting and updating the replica set.
-func (r *Range) SetDesc(desc *proto.RangeDescriptor) {
+// SetDesc atomically sets the range's descriptor. This method calls
+// ProcessRangeDescriptorUpdate() to make the range manager handle the
+// descriptor update. Note that ProcessRangeDescriptorUpdate()
+// acquires the metaLock.
```

Comment:
Ah, I see.. I misunderstood what metaLock means. Added the comment back with a TODO.